include/cuco/detail/trie/dynamic_bitset/dynamic_bitset.cuh

```diff
@@ -0,0 +1,359 @@
+   */
+  // TODO: this is not a real ref type, to be changed
+  struct storage_ref_type {
+    const slot_type* words_ref_;  ///< Words refs
+
+    const rank* ranks_ref_;         ///< Ranks refs
+    const size_type* selects_ref_;  ///< Selects refs
+
+    const rank* ranks0_ref_;         ///< Ranks refs for 0 bits
+    const size_type* selects0_ref_;  ///< Selects refs 0 bits
+  };
+
+  /**
+   * @brief Device non-owning reference type of dynamic_bitset
+   */
+  class reference {
+   public:
+    /**
+￼   * @brief Constructs dynamic_bitset_ref.
+￼   *
+￼   * @param storage Struct with non-owning refs to bitvector slot storages
+￼   */
+    __host__ __device__ explicit constexpr reference(storage_ref_type storage) noexcept;
+
+    /**
+     * @brief Access value of a single bit
+     *
+     * @param key Position of bit
+     *
+     * @return Value of bit at position specified by key
+     */
+    [[nodiscard]] __device__ constexpr bool get(size_type key) const noexcept;
+
+    /**
+     * @brief Access a single word of internal storage
+     *
+     * @param word_id Index of word
+     *
+     * @return Word at position specified by index
+     */
+    [[nodiscard]] __device__ constexpr slot_type get_word(size_type word_id) const noexcept;
+
+    /**
+     * @brief Find position of first set bit starting from a given position (inclusive)
+     *
+     * @param key Position of starting bit
+     *
+     * @return Index of next set bit
+     */
+    [[nodiscard]] __device__ size_type find_next_set(size_type key) const noexcept;
+
+    /**
+     * @brief Find number of set bits (rank) in all positions before the input position (exclusive)
+     *
+     * @param key Input bit position
+     *
+     * @return Rank of input position
+     */
+    [[nodiscard]] __device__ constexpr size_type rank(size_type key) const noexcept;
+
+    /**
+     * @brief Find position of Nth set (1) bit counting from start of bitvector
+     *
+     * @param count Input N
+     *
+     * @return Position of Nth set bit
+     */
+    [[nodiscard]] __device__ constexpr size_type select(size_type count) const noexcept;
+
+    /**
+     * @brief Find position of Nth not-set (0) bit counting from start of bitvector
+     *
+     * @param count Input N
+     *
+     * @return Position of Nth not-set bit
+     */
+    [[nodiscard]] __device__ constexpr size_type select0(size_type count) const noexcept;
+
+   private:
+    /**
+     * @brief Helper function for select operation that computes an initial rank estimate
+     *
+     * @param count Input count for which select operation is being performed
+     * @param selects Selects array
+     * @param ranks Ranks array
+     *
+     * @return index in ranks which corresponds to highest rank less than count (least upper bound)
+     */
+    template <typename SelectsRef, typename RanksRef>
+    [[nodiscard]] __device__ constexpr size_type get_initial_rank_estimate(
```

Comment:
```suggestion
    [[nodiscard]] __device__ constexpr size_type initial_rank_estimate(
```
`get_` prefixes rarely provide useful information so let's just remove it.

Reply:
59457e91c5eba5f93d45563c54336a9282f26b2b